daemon/prune.go

```diff
@@ -150,3 +154,76 @@ func (daemon *Daemon) ImagesPrune(config *types.ImagesPruneConfig) (*types.Image
 
 	return rep, nil
 }
+
+func countValidEndpoints(nw libnetwork.Network) int {
```

Comment:
I am not sure why we need this function.
I think the caller only needs to check whether `len(nw.Endpoints()==0)` when deciding if the network can be pruned.

Regardless of the count value returned by this function, if the number of endpoints is not 0, the network deletion will fail I think.